src/main/java/org/alfresco/service/namespace/QName.java

```diff
@@ -203,19 +195,42 @@ public static QName createQNameWithValidLocalName(final String namespaceURI, Str
      */
     public static String createValidLocalName(String name)
     {
-        // Validate length
-        if (name == null || name.length() == 0)
-        {
-            throw new IllegalArgumentException("Local name cannot be null or empty.");
-        }
+        checkLocalName(name);
         if (name.length() > MAX_LENGTH)
         {
             name = name.substring(0, MAX_LENGTH);
         }
 
         return name;
     }
-    
+
+    /**
+     * Condition to check string name provided for QName
+     *
+     * @param name  name to create valid local name from
+     * @throws InvalidQNameException
+     */
+    private static void checkLocalName(String name)
+        throws InvalidQNameException
+    {
+        if (name == null || name.length() == 0)
+        {
+            throw new InvalidQNameException("A QName must consist of a local name and cannot be null or empty");
+        }
+
+        if (name != null)
+        {
+            for (char illegalCharacter : illegalCharacters)
+            {
+                if (name.indexOf(illegalCharacter) != -1)
+                {
+                    throw new InvalidQNameException("name contains characters that are not permitted: "+name.charAt(name.indexOf(illegalCharacter)));
```

Comment:
The message in this exception should start from uppercase letter, also I think it should specify that it is a "local name".
I noticed that you are checking for the exact message in tests. The checks are duplicated and will have to be fixed everywhere is somebody changes the message like in this case. It is better to avoid the checks for the exact message unless it is absolutely necessary to ensure the right contents are present. I would recommend to just check for the exception of a particular class.

Reply:
Sure. Done!